### ISSUE TITLE
Product v2 specific price combination choices doesn't consider shop constraint

### ISSUE THREAD

**Issue by zuk3975** [PrestaShop/PrestaShop#30798]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When adding/editin specific price for product in product page v2 we have selection of shops and selection of combinations. 
These two options are now not dependant on each other.

### Expected behavior

Shop and combination selections should depend on each other. When shop selection changes, combinations list must be reloaded dynamically - 
1. when ALL SHOPS is selected, then combinations from all shops should be shown in choices list
2. when certain shop is selected, then combinations only from that shop should be shown in choices list

### Steps to reproduce

1. Go to product v2 (when multistore is enabled and product v2 page is enabled) and open any product with combination for editing.
2. In Pricing tab press Add specific price
3. Play around with shop selection and combination seleciton inputs and you will see that these always shows same combination no matter if single or all shops is selected (and it will cause issues when there are different combinations between shops)

### PrestaShop version(s) where the bug happened

develop

**Comment by zuk3975** [comment]
need to add shop constraint option in CombinationChoicesProvider and handle dynamic part with javascript.

**Comment by HanaRebaiQA** [comment]
Hello @zuk3975 

Thanks for your report.
It seems that your issue is similar to this one #29332 ? 
What do you Think ?

Thank You!

**Comment by zuk3975** [comment]
> Hello @zuk3975
> 
> Thanks for your report. It seems that your issue is similar to this one #29332 ? What do you Think ?
> 
> Thank You!

yes, similar

**Comment by HanaRebaiQA** [comment]
Hello @zuk3975 ,

We are aware of this issue, it is already in our debug backlog. This issue is a duplicate of #29332
To be informed when it's fixed, please subscribe to the issue mentioned above.

Thank You!

**Comment by HanaRebaiQA** [comment]
Duplicate of #29332

**Comment by zuk3975** [comment]
@HanaRebaiQA sorry for confusion - it is SIMILAR issue by its nature, but it is not the same.
If you want to close this as duplicate then add it as a "subtask" or whatever to the mentioned issue, because I don't see that one mentioning combinations anywhere. 
TLDR; 
There are 2 things to fix:
1. The customer search depending on shop (your mentioned issue #29332)
2. The combination  choices depending on shop selection (this issue)

So this should stay open, unless you mention the second part in your issue as a subtask.

**Comment by HanaRebaiQA** [comment]
Hello @zuk3975 

Don't worry :) I think if we keep your issue too and it can both be fixed in different PRs.
So, I was able to reproduce your issue.

![image](https://user-images.githubusercontent.com/92912932/212020654-76ace297-0b5a-4ae4-8598-c93a979080b1.png)

https://user-images.githubusercontent.com/92912932/212020697-cb3305d1-986a-4538-b51c-89d80f749143.mp4

Thank You!

**Comment by HanaRebaiQA** [comment]
It's related to #29332
